op-e2e/actions/proofs/jovian_activation_test.go

```diff
@@ -0,0 +1,210 @@
+)
+
+func setMinBaseFeeViaSystemConfig(t actionsHelpers.Testing, env *helpers.L2FaultProofEnv, minBaseFee uint64) {
+	// Create system config contract binding
+	systemConfig, err := bindings.NewSystemConfig(env.Sd.RollupCfg.L1SystemConfigAddress, env.Miner.EthClient())
+	require.NoError(t, err)
+
+	// Create transactor for the deployer (system config owner)
+	deployerTx, err := bind.NewKeyedTransactorWithChainID(env.Dp.Secrets.Deployer, env.Sd.RollupCfg.L1ChainID)
+	require.NoError(t, err)
+	t.Logf("Setting min base fee on L1: minBaseFee=%d", minBaseFee)
+
+	// Mine the L1 transaction
+	env.Miner.ActL1StartBlock(12)(t)
+	_, err = systemConfig.SetMinBaseFee(deployerTx, minBaseFee)
+	require.NoError(t, err, "SetMinBaseFee transaction failed")
+	env.Miner.ActL1IncludeTx(env.Dp.Addresses.Deployer)(t)
+	env.Miner.ActL1EndBlock(t)
+}
+
+func Test_ProgramAction_JovianActivation(gt *testing.T) {
+
+	runJovianDerivationTest := func(gt *testing.T, testCfg *helpers.TestCfg[any], genesisConfigFn func(*genesis.DeployConfig), jovianAtGenesis bool, minBaseFee uint64) {
+		t := actionsHelpers.NewDefaultTesting(gt)
+
+		env := helpers.NewL2FaultProofEnv(t, testCfg, helpers.NewTestParams(), helpers.NewBatcherCfg(), genesisConfigFn)
+
+		t.Logf("L2 Genesis Time: %d, JovianTime: %d ", env.Sequencer.RollupCfg.Genesis.L2Time, *env.Sequencer.RollupCfg.JovianTime)
+
+		if jovianAtGenesis {
+			// Verify Jovian is active at genesis
+			require.True(t, env.Sequencer.RollupCfg.IsJovian(env.Sequencer.RollupCfg.Genesis.L2Time), "Jovian should be active at genesis")
+		} else {
+			// If Jovian is not activated at genesis, build some blocks up to the activation block
+			// and verify that the extra data is Holocene
+			for env.Engine.L2Chain().CurrentBlock().Time < *env.Sequencer.RollupCfg.JovianTime {
+				b := env.Engine.L2Chain().GetBlockByHash(env.Sequencer.L2Unsafe().Hash)
+				expectedHoloceneExtraData := eip1559.EncodeHoloceneExtraData(250, 6)
+				require.Equal(t, expectedHoloceneExtraData, b.Extra(), "extra data should match Holocene format")
+				env.Sequencer.ActL2EmptyBlock(t)
+			}
+		}
+
+		// Build the activation block
+		env.Sequencer.ActL2EmptyBlock(t)
+		activationBlock := env.Engine.L2Chain().GetBlockByHash(env.Sequencer.L2Unsafe().Hash)
+		require.Equal(t, eip1559.EncodeJovianExtraData(250, 6, 0), activationBlock.Extra(), "activation block should have Jovian extraData")
+		require.Greater(t, activationBlock.BaseFee().Uint64(), uint64(0), "activation block should have a base fee above the minimum base fee since it was just enabled")
+
+		// Set the minimum base fee
+		setMinBaseFeeViaSystemConfig(t, env, minBaseFee)
+
+		// Allow L1->L2 derivation to propagate the SystemConfig change
+		env.Sequencer.ActL1HeadSignal(t)
+		env.Sequencer.ActL2PipelineFull(t)
+
+		// Build L2 blocks up to the L1 origin that includes the SystemConfig change
+		env.Sequencer.ActBuildToL1Head(t)
+
+		// Build activation+1 block
+		env.Sequencer.ActL2EmptyBlock(t)
```

Comment:
Is this really activation+1 block? We built several blocks during `ActBuildToL1Head(t)`, so I think we are some way past activation now.

Reply:
Could we assert on the extradata of the blocks which were past the Jovian activation, but before the L1 origin moved to the SystemConfig change? It should have a zero `minBaseFee`.